src/views/user-profile.vue

```diff
@@ -130,70 +141,52 @@
 
 <script lang="ts">
 import useProfile from "@/composable/use-profile";
-import { defineComponent, ref } from "vue";
+import { computed, defineComponent, onMounted, ref } from "vue";
+import { useRoute } from "vue-router";
 
 export default defineComponent({
   setup() {
-    const id = ref();
+    let id = ref("");
+    const route = useRoute();
     const { person, profile } = useProfile();
 
-    return { id, profile, person };
-  },
-  mounted() {
-    this.id = this.$route.params.id;
-    this.profile(this.id);
-  },
-  computed: {
-    picture: function (): string[] {
-      return this.person.map(function (person) {
-        return person.picture.large;
-      });
-    },
-    fullName: function (): string[] {
-      return this.person.map(function (person) {
-        return person.name.first + " " + person.name.last;
-      });
-    },
-    profileId: function (): string[] {
-      return this.person.map(function (person) {
-        return person.id.name + person.id.value;
+    const fullName = (firstName: string, lastName: string) =>
+      computed(() => {
+        return firstName + " " + lastName;
       });
-    },
-    firstName: function (): string[] {
-      return this.person.map(function (person) {
-        return person.name.first;
+
+    const userId = (name: string, value: string) =>
+      computed(() => {
+        return name + value;
       });
-    },
-    location: function (): string[] {
-      return this.person.map(function (person) {
+
+    const location = (
```

Comment:
you can actually access `person` in your script. No need to create a `computed` that takes parameters.